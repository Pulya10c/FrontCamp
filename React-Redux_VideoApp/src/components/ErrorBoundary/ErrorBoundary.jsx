import React from "react";
import {
  ErrorWrapper,
  ErrorTitle,
  ErrorSubTitle,
  ErrorContent
} from "./styles";

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    console.error(error);
    return { hasError: true };
  }

  componentDidCatch(error, info) {
    console.error(error);
    console.log(info);
  }

  handleClickReloadPage = () => {
    window.location.reload();
  };

  render() {
    if (this.state.hasError) {
      return (
        <ErrorWrapper>
          <ErrorTitle>Sorry!</ErrorTitle>
          <ErrorSubTitle>Something went wrong.</ErrorSubTitle>
          <ErrorContent>
            Try to <a onClick={this.handleClickReloadPage}>REFRESH</a> the page
          </ErrorContent>
        </ErrorWrapper>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
