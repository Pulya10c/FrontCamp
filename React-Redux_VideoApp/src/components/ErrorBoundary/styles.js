import styled, { css } from 'styled-components'

export const commonAnimation = css`
  opacity: 0;
  animation: appear 500ms cubic-bezier(0, 1, 0.5, 1) 1 normal forwards;

  @keyframes appear {
    0% {
      opacity: 0;
      transform: translateY(100%);
    }
    100% {
      opacity: 1;
      transform: translateY(0);
    }
  }
`

export const ErrorWrapper = styled.div`
  margin: 40px auto;
  text-align: center;
`

export const ErrorTitle = styled.h1`
  ${commonAnimation}

  font-size: 40px;
  animation-delay: 0;
`

export const ErrorSubTitle = styled.h2`
  ${commonAnimation}

  font-size: 30px;
  animation-delay: 200ms;
`

export const ErrorContent = styled.p`
  ${commonAnimation}

  font-size: 18px;
  margin: 2em;
  animation-delay: 500ms;
`
