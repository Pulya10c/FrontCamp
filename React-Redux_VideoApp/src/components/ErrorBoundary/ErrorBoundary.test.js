import React from 'react';
import { shallow } from 'enzyme';
import ErrorBoundary from './ErrorBoundary';

describe('Component ErrorBoundary', () => {
  let wrapper;
  const Child = () => <div>child</div>;
  beforeEach(() => {
    wrapper = shallow(
      <ErrorBoundary>
        <Child />
      </ErrorBoundary>
    );
  });

  it('should render Child correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render ErrorWrapper correctly', () => {
    wrapper.setState({ hasError: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should call window.location.reload if to call handleClickReloadPage', () => {
    window.location.reload = jest.fn();
    wrapper.instance().handleClickReloadPage();
    expect(window.location.reload).toHaveBeenCalled();
  });
});
