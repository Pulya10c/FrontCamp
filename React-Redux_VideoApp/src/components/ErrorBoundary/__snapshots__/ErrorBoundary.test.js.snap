// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ErrorBoundary should render Child correctly 1`] = `<Child />`;

exports[`Component ErrorBoundary should render ErrorWrapper correctly 1`] = `
<styled.div>
  <styled.h1>
    Sorry!
  </styled.h1>
  <styled.h2>
    Something went wrong.
  </styled.h2>
  <styled.p>
    Try to 
    <a
      onClick={[Function]}
    >
      REFRESH
    </a>
     the page
  </styled.p>
</styled.div>
`;
